### DISCUSSION TITLE
Pre-loaded chapters get loaded in a zero-height/zero-width webView

### DISCUSSION THREAD

**Discussion by smoores-dev** [readium/kotlin-toolkit#513] [Q&A]
Hi! I... am not sure where to start here; I've been messing around with this for quite a while and I'm very much at a dead end.

I'll start by saying that I haven't opened an issue because I fully can't reproduce this on the test app, so this must be in some way something that I'm doing.

In [Storyteller](https://gitlab.com/smoores/storyteller/-/blob/main/mobile), when I first load the navigator, it correctly loads the current chapter, previous chapter, and next chapter. However, when I swipe through to the next chapter, the _next_ pre-loaded chapter's webview has height and width 0. This happens in the other direction as well; basically any time I navigate to a chapter that wasn't in the initial three loaded when the view is first loaded.

I'm not sure what's even helpful, but here's my Fragment that uses the EpubNavigatorFactory: https://gitlab.com/smoores/storyteller/-/blob/main/mobile/modules/readium/android/src/main/java/expo/modules/readium/EpubFragment.kt?ref_type=heads

And here's the View that renders that Fragment: https://gitlab.com/smoores/storyteller/-/blob/main/mobile/modules/readium/android/src/main/java/expo/modules/readium/EpubView.kt?ref_type=heads

**Comment by mickael-menu** [comment]
I can't find any issue with the code you shared. Maybe an interoperability issue with Expo? I'm not familiar with React Native, but maybe you would have a better chance with https://github.com/5-stones/react-native-readium?

**Reply by smoores-dev** [reply]
Thanks for taking a look, @mickael-menu, I really appreciate it! That library unfortunately doesn't expose much of the functionality that I need to implement Media Overlays :/. Honestly it's a bit of a relief that you don't see anything obviously wrong with that Android code! I'm by no means an Android expert but I really did think I was doing it correctly haha.

Would you be able to point me to where in the kotlin-toolkit the offscreen/pre-loaded web views get created? I'm curious to see if I can maybe set some breakpoints or something to try to understand why they're not sized correctly

**Reply by mickael-menu** [reply]
Preloading is handled by an instance of [`ViewPager`](https://github.com/readium/kotlin-toolkit/blob/091f9ac0638678987449f1b2097daa9c0c6df8fa/readium/navigator/src/main/java/org/readium/r2/navigator/pager/RtlViewPager.java#L43). The default settings are used, [which is to preload one more resource](https://developer.android.com/reference/android/support/v4/view/ViewPager#setOffscreenPageLimit%28int%29).
